### ISSUE TITLE
Weird Error/stuck behavior, that start to come out with force function - NetFramework / CAFE4242 stub

### ISSUE THREAD

**Issue by Special-Niewbie** [Winetricks/winetricks#1784]
Hello everyone, I don't know if it's me or something else but it starts to come out this weird error. On fresh installation of Ubuntu 2104 -Wine - Winetricks , With Winetricks as soon I will start to install in force mode NetFramework 4.7.2 that the installer will install all the NetFramework from 4.0/ 4.5/ 4.5.2 and so on till 4.7.2 few weeks ago I can install all of them but now with another new fresh installation of Ubuntu 21.04 it come out that if I launch the `winetricks --force dotnet472` command, it start to install the 4.0 and when need to start the next 4.5 or 4.5.2 or 4.6 (the error stuck will be manifest around these versions) and cannot be able to continue the installation (I left even one entire night because i know that I need to wait about). I post two example error that will stuck always in this voice of `DeregisterEventSource CAFE4242`, do you know how to solve this?

![Schermata del 2021-06-23 01-25-49](https://user-images.githubusercontent.com/64843123/123032131-09951180-d418-11eb-8e9c-01f2a2262101.png)

![Schermata del 2021-06-23 11-26-48](https://user-images.githubusercontent.com/64843123/123032145-0e59c580-d418-11eb-839a-a614c936e5d8.png)

**Comment by Special-Niewbie** [comment]
**Edit 2:** For your infos, it seems that the Reboot system requested by the installer - once you finished to install one Version of NetFramework - it hangs the back ground process Wine System ( and to my case in this voice `DeregisterEventSource CAFE4242` ) seems for that reason it doesn't continue with the next installation. If I will install all each of single Framework Version separately, it will install them,  but if I'm lucky I can continue to install the next strict away otherwise it needs a real all System Linux Reboot to continue with the next version of Net Framework installation. I don't know if it's kind of a problem of Winetricks or Wine with them related new libraries that let it happened this .....

**Comment by austin987** [comment]
This is almost certainly an upstream wine issue. I don't know of a workaround (other than downgrading to a working wine version).

**Comment by Special-Niewbie** [comment]
> This is almost certainly an upstream wine issue. I don't know of a workaround (other than downgrading to a working wine version).

Thank you to reply @austin987 , for luck still the goes around and it's to install all each the NET.Framework from 4.0 to 4.7.2 and reboot each time or close winetricks each time and re-open it, but trust me it took me few hours.... I believe it's related to some upgrade of the libs that Wine was received these past days , because for a couple of days Wine in fresh Ubuntu installation was broken, caused of some blocked libs that they adjusted....
I just get you inform that there is this kind of problem , after ( maybe ) when Wine will be upgrade again , maybe will disappear.

**Comment by Chiitoo** [comment]
You might like trying terminating all Wine processes manually if they get stuck (for example via using 'wineserver -k') instead of a full reboot, which really should not be necessary unless something somewhere outside of Wine is going very wrong.

**Comment by Special-Niewbie** [comment]
> You might like trying terminating all Wine processes manually if they get stuck (for example via using 'wineserver -k') instead of a full reboot, which really should not be necessary unless something somewhere outside of Wine is going very wrong.

@Chiitoo , honestly I didn't know and tried about this command `wineserver -k` thank you about, but one thing that I noticed and Ubuntu 21.04 it have this problem, sometimes (not every time) during the installation, Ubuntu system show one pop-up related to the error `wine-stable-i386 not installed ` but in fact it's installed and well set. I noticed this with the Net Framework . And it's not my system or the way how I installed everything, but it have a kind of this little bug problem (not serious) that with Net-Framework I noticed, I don't know if could be even this that put in crisis Winetricks automated Net.Framework installation. 

Lastly, there is way to accelerate the Net.framework Installation, because it took even few hours before install all of them or it's how Wine manage in buggy way these Net Framework installation? Because it use a Single Thread/Core and also in buggy way. This it happen only with Net.Framework otherwise the app or even games works with full Thread/Cores.
